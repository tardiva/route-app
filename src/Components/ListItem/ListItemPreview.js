import React, {Component} from 'react';
import Icon from 'react-icons-kit';
import {ic_close} from 'react-icons-kit/md/ic_close';

class ListItemPreview extends Component {

  render() {
    return (
      <div className="list-item" style={this.props.style}>
        <div className="list-item__content">{this.props.name}</div>
        <button className="list-item__delete-button">
          <Icon icon={ic_close}/>
        </button>
      </div>)
  }
}

export default ListItemPreview;