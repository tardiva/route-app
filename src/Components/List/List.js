import React, { Component } from 'react';
import './List.css';
import { DragDropContext } from 'react-dnd';
import MultiBackend, { Preview } from 'react-dnd-multi-backend';
import HTML5toTouch from 'react-dnd-multi-backend/lib/HTML5toTouch';
import ListItem from '../ListItem/ListItem';
import ListItemPreview from '../ListItem/ListItemPreview'

export class List extends Component {

  render() {
    return (
      <div className="list">
        <ul className="list__list">
          {this.props.items.map((item, i) => {
            return (<ListItem key={item.id}
                              index={i}
                              item={item}
                              moveItem={this.props.moveItem}
                              removeItem={this.props.removeItem}
            />)
          })}
        </ul>
        <Preview generator={this.generatePreview} />
      </div>
    );
  }

  generatePreview(type, item, style) {
    Object.assign(style, {width: `${item.width}px`});
    return (<ListItemPreview name={item.name} style={style} />);
  }
}

export default DragDropContext(MultiBackend(HTML5toTouch))(List);