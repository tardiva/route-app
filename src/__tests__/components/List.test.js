import React from 'react';
import { shallow } from 'enzyme';
import List from '../../Components/List/List';
import points from '../../Fixtures/points';
import ListItemPreview from '../../Components/ListItem/ListItemPreview';

// Obtain the reference to the component before React DnD wrapping
const OriginalList = List.DecoratedComponent;

it('should render List correctly', () => {
  const wrapper = shallow(<OriginalList items={points} />);
  expect(wrapper).toMatchSnapshot();
});

it('should render empty List', () => {
  const wrapper = shallow(<OriginalList items={[]} />);
  expect(wrapper).toMatchSnapshot();
});

it('should generate ListItemPreview', () => {
  const wrapper = shallow(<OriginalList items={[]} />);
  const type = {};
  const item = {name: 'Test', width: 300};
  const style = {};
  const preview = shallow(wrapper.instance().generatePreview(type, item, style));
  expect(preview).toMatchSnapshot();
});