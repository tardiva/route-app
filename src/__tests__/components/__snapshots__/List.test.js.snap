// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate ListItemPreview 1`] = `
<div
  className="list-item"
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="list-item__content"
  >
    Test
  </div>
  <button
    className="list-item__delete-button"
  >
    <Icon
      fill="currentColor"
      icon={
        Object {
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z",
              },
              "name": "path",
            },
          ],
          "viewBox": "0 0 24 24",
        }
      }
      size={16}
    />
  </button>
</div>
`;

exports[`should render List correctly 1`] = `
<div
  className="list"
>
  <ul
    className="list__list"
  >
    <DropTarget(DragSource(ListItem))
      index={0}
      item={
        Object {
          "coord": Object {
            "lat": 0,
            "lng": 0,
          },
          "id": 0,
          "name": "My first point",
        }
      }
      key="0"
    />
    <DropTarget(DragSource(ListItem))
      index={1}
      item={
        Object {
          "coord": Object {
            "lat": 1,
            "lng": 1,
          },
          "id": 1,
          "name": "My second point",
        }
      }
      key="1"
    />
    <DropTarget(DragSource(ListItem))
      index={2}
      item={
        Object {
          "coord": Object {
            "lat": 2,
            "lng": 2,
          },
          "id": 2,
          "name": "My third point",
        }
      }
      key="2"
    />
  </ul>
  <Preview
    generator={[Function]}
  />
</div>
`;

exports[`should render empty List 1`] = `
<div
  className="list"
>
  <ul
    className="list__list"
  />
  <Preview
    generator={[Function]}
  />
</div>
`;
